roles/setup/tasks/setup-container.yml

```diff
@@ -20,25 +20,66 @@
     XDG_RUNTIME_DIR: "/run/user/{{ receptor_user_register.uid }}"
   vars:
     receptor_uid_directory: "/run/user/{{ receptor_user_register.uid }}"
+    receptor_image_with_digest: "{{ receptor_image_registry }}/{{ receptor_image_name }}@{{ receptor_image_sha }}"
   block:
+    - name: Set facts
+      ansible.builtin.set_fact:
+        user_home_directory: "/home/{{ receptor_user }}"
+
+    - name: Create containers.conf file
+      ansible.builtin.template:
+        src: templates/containers.conf.j2
+        dest: ~/.config/containers/containers.conf
+        mode: '0750'
+        owner: "{{ receptor_user }}"
+        group: "{{ receptor_group }}"
+
     - name: Enable podman socket
       ansible.builtin.systemd:
         name: podman.socket
         state: started
         enabled: true
         scope: user
 
+    - name: Login to registry if authentication is provided
+      when: (receptor_image_registry_username is defined and receptor_image_registry_token is defined) or receptor_image_authfile_path is defined
+      block:
+        - name: Login to registry with username and token
+          when: receptor_image_registry_username is defined and receptor_image_registry_token is defined
+          containers.podman.podman_login:
+            username: "{{ receptor_image_registry_username }}"
+            password: "{{ receptor_image_registry_token }}"
+            registry: "{{ receptor_image_registry }}"
+
+        - name: Copy authfile with owner and permissions
+          when: receptor_image_authfile_path is defined
+          ansible.builtin.copy:
+            src: "{{ receptor_image_authfile_path }}"
+            dest: ~/.config/containers/auth.json
+            owner: "{{ receptor_user }}"
+            group: "{{ receptor_group }}"
+            mode: '0644'
+
+        # Username and password are required values, but authentication is managed through the authfile
+        - name: Login to registry with authfile
+          when: receptor_image_authfile_path is defined
+          containers.podman.podman_login:
+            authfile: ~/.config/containers/auth.json
+            registry: "{{ receptor_image_registry }}"
+            username: ''
+            password: ''
+
+    # Using command module instead of podman_image module here,
+    # as the podman_image module does not yet support pulling by digest
     - name: Pull the latest release of receptor
-      containers.podman.podman_image:
-        name: "{{ receptor_image_name }}"
-        tag: "{{ receptor_image_tag }}"
-        auth_file: "{{ lookup('ansible.builtin.file', receptor_image_authfile, errors='ignore') }}"
-        state: present
+      ansible.builtin.command:
+        cmd: "podman pull {{ receptor_image_with_digest }}"
+      changed_when: false
```

Comment:
FYI pulling via digest is supported by the `podman_image` module since 1.10.2

https://github.com/containers/ansible-podman-collections/commit/d5f7655464ed942a368e317f99e39bc14acf0794

Reply:
Thank you Dimitri, 
I had tried to configure this before with the `name` field, but will try this again to see if I can get this working with the example you provided

Reply:
I was able to try it successfully with

```console
$ ansible -i localhost, all -c local -m containers.podman.podman_image -a 'name=quay.io/ansible/receptor@sha256:75a7dade4d9dac4a52edc24eaf80d2c8ffdcdbd1bfa5f0cbcb51ff88284c5aea'
localhost | CHANGED => {
    "actions": [
        "Pulled image quay.io/ansible/receptor@sha256:75a7dade4d9dac4a52edc24eaf80d2c8ffdcdbd1bfa5f0cbcb51ff88284c5aea"
    ],
(...)
    "podman_actions": [
        "/usr/bin/podman image ls quay.io/ansible/receptor@sha256:75a7dade4d9dac4a52edc24eaf80d2c8ffdcdbd1bfa5f0cbcb51ff88284c5aea --format json",
        "/usr/bin/podman image exists quay.io/ansible/receptor@sha256:75a7dade4d9dac4a52edc24eaf80d2c8ffdcdbd1bfa5f0cbcb51ff88284c5aea",
        "/usr/bin/podman pull quay.io/ansible/receptor@sha256:75a7dade4d9dac4a52edc24eaf80d2c8ffdcdbd1bfa5f0cbcb51ff88284c5aea -q",
        "/usr/bin/podman inspect 23f0a37e60f672d524b035f789e4c5fe9ec3c73a044932492938a1ead2daa0ae --format json"
    ],
    "stdout": "",
    "stdout_lines": []
}
```

Reply:
@dsavineau This is what I initially saw  when trying to use the module when pulling w/ SHA - 

```
TASK [ansible.receptor.setup : debug] *****************************************************************************************************************************************
ok: [remote-execution] => {
    "receptor_image_with_digest": "quay.io/zkayyali812/awx-ee@sha256:d6e2920adf915c5fd819956219e630fdf7e0813c27b3808af84d6b8be7409641"
}

TASK [ansible.receptor.setup : Pull image with SHA256 tag] ********************************************************************************************************************
fatal: [remote-execution]: FAILED! => {"changed": false, "msg": "Failed to pull image quay.io/zkayyali812/awx-ee:sha256:d6e2920adf915c5fd819956219e630fdf7e0813c27b3808af84d6b8be7409641"}
```

With this task -
```
- name: Pull image with SHA256 tag
      containers.podman.podman_image:
        name: "{{ receptor_image_with_digest }}"
        state: present
```

Reply:
Turns out I was on version 1.10.1. After updating to 1.11, this is working as expected! Thank you

Reply:
note that you'll need to reflect this requirements in the `galaxy.yml` file

Reply:
I just updated to add this. Was wondering if this should instead be in requirements.yml file, but I added it to the galaxy.yml file.